lib/models/graph-object.js

```diff
@@ -43,6 +43,38 @@ function GraphModelFactory (Model, Constants, configuration) {
             active: function() {
                 var obj = this.toObject();
                 return Constants.Task.ActiveStates.indexOf(obj._status) > -1;
+            },
+            toJSON: function () {
+                var workflow = this.toObject();
+                var sanitizedWorkflow = workflow;
+                var walked = [];
+                var stack = [ { workflow : workflow , stack : '' } ];
+                while ( stack.length > 0 ) {
+                    var item = stack.pop();
+                    var workflow  = item.workflow;
+                    for ( var property in workflow ) {
+                        if ( workflow.hasOwnProperty( property )) {
+                            if ( typeof workflow[ property ] === "object" ) {
+                                var alreadyFound = false;
+                                for ( var i = 0; i < walked.length; i++ ) {
+                                    if ( walked[i] === workflow[ property ] ) {
+                                        alreadyFound = true;
+                                        break;
+                                    }
+                                }
+                                if ( !alreadyFound ) {
+                                    walked.push( workflow[ property ]);
+                                    stack.push( { workflow : workflow[ property ], stack: item.stack + '.' + property });
+                                }
+                            } else {
+                                if ( property.match( /password/i )) {
+                                    workflow[property] = '*** REDACTED ***';
```

Comment:
I hope we can use the same redacted word, see: https://github.com/RackHD/on-core/blob/master/lib/serializables/log-event.js#L193